crates/runtime/src/datafusion.rs

```diff
@@ -193,7 +199,38 @@ impl DataFusion {
             }
         });
 
-        self.tasks.push(task_handle);
+        self.connectors_tasks.insert(table_name, task_handle);
+
+        Ok(())
+    }
+
+    #[must_use]
+    pub fn table_exists(&self, dataset_name: &str) -> bool {
+        self.ctx.table_exist(dataset_name).unwrap_or(false)
+    }
+
+    pub fn remove_table(&mut self, dataset_name: &str) -> Result<()> {
+        let table_exists = self.ctx.table_exist(dataset_name).unwrap_or(false);
+        if !table_exists {
+            return TableDoesNotExistSnafu.fail();
+        }
```

Comment:
I don't think we should error out here - if we want to remove something that doesn't exist, we just return success without doing anything.

Reply:
Updated.